capsules/src/virtual_rng.rs

```diff
@@ -0,0 +1,169 @@
+            ReturnCode::SUCCESS
+        }
+    }
+}
+
+impl<'a> Client for MuxRngMaster<'a> {
+    fn randomness_available(
+        &self,
+        _randomness: &mut dyn Iterator<Item = u32>,
+        _error: ReturnCode,
+    ) -> Continue {
+        // Try find if randomness is available, or return done
+        self.inflight.take().map_or(Continue::Done, |device| {
+            self.do_next_op();
+            let cont_code = device.randomness_available(_randomness, _error);
+
+            // Get more randomness if requested
+            if cont_code == Continue::More {
+                device.get();
+            }
+
+            cont_code
+        })
+    }
+}
+
+// Struct for a single rng device
+pub struct VirtualRngMasterDevice<'a> {
+    //reference to the mux
+    mux: &'a MuxRngMaster<'a>,
+
+    // Pointer to next element in the list of devices
+    next: ListLink<'a, VirtualRngMasterDevice<'a>>,
+    client: OptionalCell<&'a dyn Client>,
+    operation: Cell<Op>,
+}
+
+// Implement ListNode trait for virtual rng device
+impl<'a> ListNode<'a, VirtualRngMasterDevice<'a>> for VirtualRngMasterDevice<'a> {
+    fn next(&self) -> &'a ListLink<VirtualRngMasterDevice<'a>> {
+        &self.next
+    }
+}
+
+impl<'a> VirtualRngMasterDevice<'a> {
+    pub const fn new(mux: &'a MuxRngMaster<'a>) -> VirtualRngMasterDevice<'a> {
+        VirtualRngMasterDevice {
+            mux: mux,
+            next: ListLink::empty(),
+            client: OptionalCell::empty(),
+            operation: Cell::new(Op::Idle),
+        }
+    }
+}
+
+impl<'a> PartialEq<VirtualRngMasterDevice<'a>> for VirtualRngMasterDevice<'a> {
+    fn eq(&self, other: &VirtualRngMasterDevice<'a>) -> bool {
+        // Check whether two rng devices point to the same device
+        self as *const VirtualRngMasterDevice<'a> == other as *const VirtualRngMasterDevice<'a>
+    }
+}
+
+impl<'a> Rng<'a> for VirtualRngMasterDevice<'a> {
+    fn get(&self) -> ReturnCode {
+        self.operation.set(Op::Get);
+        self.mux.do_next_op()
+    }
+
+    fn cancel(&self) -> ReturnCode {
+        self.mux.inflight.map_or_else(
+            || {
+                // If no node inflight, just set node to idle and return
+                self.operation.set(Op::Idle);
+                ReturnCode::SUCCESS
+            },
+            |current_node| {
+                // Find if current device is the one in flight or not
+                if *current_node == self {
+                    self.mux.rng.cancel()
+                } else {
+                    self.operation.set(Op::Idle);
+                    ReturnCode::SUCCESS
+                }
+            },
+        )
+    }
+
+    fn set_client(&'a self, client: &'a dyn Client) {
+        self.mux.devices.push_head(&self);
+        self.client.set(client);
+    }
+}
+
+impl<'a> Client for VirtualRngMasterDevice<'a> {
+    fn randomness_available(
+        &self,
+        randomness: &mut dyn Iterator<Item = u32>,
+        _error: ReturnCode,
```

Comment:
parameter names should only be prefixed with an underscore when they are not used in the function. error is used, so it should just be named `error` rather than `_error`